core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -61,18 +63,27 @@ public static TableMetadata newTableMetadata(TableOperations ops,
                                                PartitionSpec spec,
                                                String location,
                                                Map<String, String> properties) {
-    return newTableMetadata(schema, spec, location, properties, DEFAULT_TABLE_FORMAT_VERSION);
+    return newTableMetadata(schema, spec, SortOrder.unsorted(), location, properties, DEFAULT_TABLE_FORMAT_VERSION);
```

Comment:
This class might be a good one for a builder as well...

Reply:
Created #1412.